src/fsharp/FSharp.Core/array.fs

```diff
@@ -1270,6 +1270,96 @@ namespace Microsoft.FSharp.Collections
                 let count' = Operators.min count len
                 Microsoft.FSharp.Primitives.Basics.Array.subUnchecked 0 count' array
 
+        [<CompiledName("RemoveAt")>]
+        let removeAt (index: int) (source: 'T[]) : 'T[] =
+            checkNonNull "source" source
+            if index < 0 || index >= source.Length then invalidArg "index" "index must be within bounds of the array"
```

Comment:
The current spec in https://github.com/dotnet/fsharp/pull/11772#issuecomment-886225737 says "if it's outside the range it is ignored".  However I'm open to discussion on this

Reply:
I thought about this and if my code makes an operation that doesn't make sense I'd like to know about it. I checked and that's consistent with other .NET collections.